packages/babel-parser/src/parser/statement.js

```diff
@@ -854,6 +863,7 @@ export default class StatementParser extends ExpressionParser {
     directives: ?(N.Directive[]),
     topLevel: boolean,
     end: TokenType,
+    afterBlockParse?: Function,
```

Comment:
Can you refine this type annotation? (Also in the other functions)

```suggestion
    afterBlockParse?: (hasStrictModeDirective: boolean) => void,
```